homework-tips.md

```diff
@@ -0,0 +1,83 @@
+base is not for the Code-The-Dream-School repository, because if it is, you will not be able to merge
+your branch after the review is complete.  The head repository for the pull request must be your lesson
+branch.  Once you have created the pull request, click on Files changed.  You should see the changes
+you made.  Then click on pull requests again, and copy the link for the pull request you just made.
+This is what you submit using the homework submission form.
+
+Do NOT merge your pull request yet!  Your assignment must be reviewed.  Your reviewer may request changes.
+The reviewer will make comments on the pull request.
+If changes are necessary, you make them to the same lesson branch.  Then you add, commit, and push
+your changes.  The additional changes are automatically added to the same pull request.
+Once your reviewer has approved your lesson pull request, THEN you merge the pull request.
+
+## Creativity and Experimentation
+
+Once you have finished the assignment, you can be creative and make variations on the instructions
+and what the instructor shows you. Through creative experimentation, 
+you will learn much more than you would if you had just
+followed instructions. Of course, there is also a risk of changing so much that you get 
+confused, overwhelmed, and maybe stuck, so for safety, you do your experimentation in a special branch.
+For example, if you have just completed week 1, while the week1 branch is active, you could do:
+```
+git branch -b week1plus
+```
+Then, try some things! Once you are done experimenting, add and commmit your changes.  You don't
+need to push them to github. Again, creativity is crucial for learning. It also allows you to take full advantage of the resources CTD provides. Can't figure something out? Bring your question to a review session! Not sure if you did something the right way? Your assignment reviewer will tell you! 
+
+Subsequent lessons build on the previous ones, so you should checkout
+the week1 branch before creating the lesson branch for week2.
+
+## Recovering From a Common Mistake
+
+Suppose you forget and start working on a lesson before you create the new branch.  If you haven't yet committed your changes, you can do
+```
+git stash
+git checkout -b lesson-branch
+git stash apply
+```
+Now suppose you forget to create your new branch and you do a commit to main or master.  Don't push it to github!  Go ahead and add and commit any additional changes.  Then do:
+```
+git checkout -b lesson-branch
+git checkout main    (or master -- it's main for newer repositories)
+git reset --hard origin/main
+git checkout lesson-branch
+```
+Now suppose you push your main or master branch containing your lesson changes to github.  Well, that is bad news.  It is not easy to recover the original state, although your mentor can help you do this.  Never push
+to the main or master branch!  This is a very bad practice, which is not allowed in real development projects.
```

Comment:
Should we specifically have them set `main`  as a protected branch in Github to prevent this from happening accidentally? (Still worth explaining to them the _why_ behind that though :+1: )